Assets/Vimeo/Scripts/Recorder/VimeoPublisher.cs

```diff
@@ -114,23 +118,38 @@ private void VideoUpdated(string response)
             UploadProgress("SaveInfoComplete", 1f);
         }
 
+        private void NetworkError(string response)
+        {
+            if (OnUploadFail != null) {
+                OnUploadFail("It seems like you are not connected to the internet, or having connection problems which disables the uploading of the video.");
+            }
+        }
+
         private void ApiError(string response)
         {
             JSONNode json = JSON.Parse(response);
 
             if (json["invalid_parameters"] != null) {
+
                 for (int i = 0; i < json["invalid_parameters"].Count; i++) {
                     // TODO use .Value
                     if (json["invalid_parameters"][i]["field"].ToString() == "\"privacy.download\"") {
-                        Debug.LogError("You must upgrade your Vimeo account in order to disable downloads on your video. https://vimeo.com/upgrade");
+                        if (OnUploadFail != null) {
+                            OnUploadFail("You must upgrade your Vimeo account in order to access this privacy feature. https://vimeo.com/upgrade");
```

Comment:
If `OnUploadFail` is not implemented, then the user will never see these messages, right? Need a way to log these out regardless of the event. I see you are logging errors in `VimeoRecorder.cs` but what about `VimeoPlayer.cs`?

Reply:
I think it would make sense to have a `virtual` method in the `VimeoAPI` class that both the publisher and player can override but it will handle the logging for everyone, in the meantime I am just straight up logging it in the `VimeoPlayer` -> https://github.com/vimeo/vimeo-unity-sdk/pull/28/commits/0fa27ea469d9ab52374d009097d8c70c475f87d3

Reply:
yes, that sounds good